inference/worker/chat_chain_prompts.py

```diff
@@ -0,0 +1,54 @@
+V2_ASST_PREFIX = "<|assistant|>"
+V2_PROMPTER_PREFIX = "<|prompter|>"
+
+ASSISTANT_PREFIX = "Open Assistant"
+HUMAN_PREFIX = "Human"
+OBSERVATION_SEQ = "Observation:"
+
+# Adjust according to the training dates and datasets used
+# Overall, Assistant is a powerful tool that can help with a wide range of tasks and provide valuable insights and information on a wide range of topics.
+# Assistant is constantly learning and improving, and its capabilities are constantly evolving.
+KNOWLEDGE_DATE_CUTOFF = "2021-09-01"
```

Comment:
Yes, I probably should adjust that

Reply:
I'm not sure we should adjust this. The model's broad knowledge will cut-off where its pretraining set did. Any knowledge beyond that date (introduced by SFT) will be very narrow and patchy at best

Reply:
Yes, that I am aware of, and there is no clear way to test it!

But just to explain what that KNOWLEDGE_DATE_CUTOFF is used for:
It helps model to decide to use a tool, for any recent data queried, but its not super accurate or important.(models are not that good, to take advantage of this info always...)

So it can be left as "2021-09-01" or changed to whatever newer date, it will not hinder my logic.
It can hinder if we put like "1921-01-01" or something like that :)

Reply:
Also can be deleted altogether, but I think it helps a little bit :)
Prompting is tricky business as you all are aware :)

Reply:
> Yes, that I am aware of, and there is no clear way to test it!
> 
> But just to explain what that KNOWLEDGE_DATE_CUTOFF is used for: It helps model to decide to use a tool, for any recent data queried, but its not super accurate or important.(models are not that good, to take advantage of this info always...)
> 
> So it can be left as "2021-09-01" or changed to whatever newer date, it will not hinder my logic. It can hinder if we put like "1921-01-01" or something like that :)

Makes sense. It would make sense to me if the model avoids relying on internal knowledge for anything beyond the cut-off for the pretraining data

Reply:
It would be perfect :)

Reply:
Here is example where correctly chooses to use external help:
<img width="934" alt="Screenshot 2023-04-30 at 11 51 14" src="https://user-images.githubusercontent.com/13547364/235369036-b452e749-bf58-4548-b10d-9035014ebdb5.png">

And then external sources failed :)